### PR TITLE
Issue #3275943 by alex.ksis: Reset cache of the commented entity.

### PR DESCRIPTION
**Pull request by ksis**
## Problem
Moderators do not see new comments in a discussion until cache is flushed.

## Solution
Reset discussion cache after a new comment was posted.

## Issue tracker
- https://www.drupal.org/project/social/issues/3275943
- https://getopensocial.atlassian.net/browse/YANG-7566

## How to test
- [x] Create a discussion with manual comments approval, add a regular user as a moderator.
- [x] Open the discussion as the moderator, make sure there are no comments.
- [x] As a CM+ post a comment, make sure it is unpublished.
- [x] Open the discussion as the moderator, make sure the created comment is not displayed.
- [x] Clear cache, verify that the comment appeared.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2889-ljd9sxz82ojiotllxmecihvfk2kze7dk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/625e8ffb2e91387b17effd8f

File: modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php
Comment (Ressinel) [inline modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: Hi, @ksis 

Since in the d.org [issue](https://www.drupal.org/project/social/issues/3275943#comment-14485864) you commented that it is a temporary solution could you add @todo, please? 
And also add more description of the problem to the commit message.

Thanks!

Comment (ksis) [reply modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: Hi @Ressinel,

I've updated the PR. Please, check if it would be enough. 

Thanks!

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2889-z69pkkkfxv0zu7rnofa7m7fudn7gevwz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62a88e7314d98b2deffe0c2f

**Review by ribel — APPROVED** [review]
Changes looks good to me.

**Review by Ressinel — APPROVED** [review]

File: modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php
Comment (navneet0693) [inline modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: Do we know why this is needed? Inline documentation should explain it better here.

Do we have a plan or follow-up for this to do? If not, I would request you to please debug and add a proper solution here.

Comment (Ressinel) [reply modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: @ksis Could you check @navneet0693 comment, please?
cc @ribel

Comment (ribel) [reply modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: @navneet0693 @Ressinel We discussed this case internally with our whole team.
Unfortunately we already spend few days on fixing this one bug and we don't have ability to continue work on it, so we didn't create a follow up. Also "add a proper solution here" is a bit vague suggestion, so we can't estimate and plan it. 
 
If adding better description to the ToDo will unblock this issue we can add it.
But if not - then we can leave it as is.
If Open Social core team can continue work on it that would be great.
If not - then you can close the PR. 
cc @tbsiqueira @bramtenhove

Comment (navneet0693) [reply modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: @ribel I agree, my suggestion is a bit vague. I will try to investigate this in the coming days and come up with a proper suggestion or fix. If this is blocking you/the team, please feel free to merge it.

Comment (ribel) [reply modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: Thanks @navneet0693. it's not blocking our team, and since this is a small change in code we an use it as a patch.
But would be great if you can investigate it when you have a time.

Comment (navneet0693) [reply modules/custom/social_ajax_comments/src/Controller/AjaxCommentsController.php:184]: @ribel Sure, Taras. I will take over this PR.

**Comment by ronaldtebrake** [comment]
I've tried to reproduce this also with discussions moderation enabled, for me it works perfectly fine now, but it seems it's not an issue of ajax comments / Distro nor an issue that I can reproduce; hopefully this means the actual root cause has been fixed in the meantime. Unfortunately couldn't find any related issue so feel free to re-open.

Screenshot 2023-03-31 at 09 28 00

**Pull request closed without merging**
